### DISCUSSION TITLE
Unable to alter/delete entry in system.users

### DISCUSSION THREAD

**Discussion by jmunsterman** [ClickHouse/ClickHouse#59593] [Q&A]
I am not sure how I did this but I seem to have backed myself into a corner. We authenticate users against our Google directory but somehow I messed up an entry. All users have storage set to 'local_directory' and auth_type to 'ldap' except for one account which has storage set to 'ldap' as well. Any attempt to remove or alter that user results in a error like:
` Cannot remove user `XXXXXX` from ldap because this storage is readonly. (ACCESS_STORAGE_READONLY)`
How can we delete (or fix) that account, short of granting write access to the directory?
Thanks!

**Comment by UnamedRus** [comment]
> from ldap because this storage is readonly. (ACCESS_STORAGE_READONLY)
> one account which has storage set to 'ldap' as well. 

All users defined via SQL?

**Comment by jmunsterman** [comment]
Yes, the users are all defined is system.users. The users that work have storage set to local_directory, but the one user somehow got set to storage 'ldap' and it won't let us change it.

**Comment by UnamedRus** [comment]
>  the users are all defined is system.users. 

it's not an answer to my question. even XML users are getting into system.users

**Reply by jmunsterman** [reply]
Maybe I misunderstood the question, all the users have been added via the client, none are defined in users.xml.

**Comment by louwar** [comment]
Hello,
I installed clickhouse and configured authentication by LDAP.
I tested it before to create any users and the login/authentication worked fine,
but right now I have some users (used for test) that I cannot modify neither delete,
because they have in the system.users, storage=ldap and it is impossible to do anything.

I can create user with same name via user.xml or via SQL, but they would have storage=user_xml or local_directory and could be manipulated,
but my issue is the same of this discussion:

Unable to alter/delete entry in system.users (specially if they are storage=ldap, auth_type=ldap

It is not normal that I cannot remove them.

Thanks & Regards
Lou
